src/ucp/api/ucp_def.h

```diff
@@ -386,12 +386,20 @@ typedef void (*ucp_tag_recv_callback_t)(void *request, ucs_status_t status,
  *
  * The enumeration allows specifying which events are expected on wakeup, though
  * empty events are possible.
+ *
+ * @note UCP doesn't generate POLLOUT like events (see poll man pages) since
+ * it's always ready to initiate outgoing operations (e.g any type of sending,
+ * atomic or RMA). Hoverer, they can be pended due to limitation of resources.
+ * It's user's responsibility to do progressing calling @ref ucp_worker_progress
+ * when it's needed to achieve the best performance.
  */
 typedef enum ucp_wakeup_event_types {
     UCP_WAKEUP_RMA         = UCS_BIT(0), /**< Remote memory access send completion */
     UCP_WAKEUP_AMO         = UCS_BIT(1), /**< Atomic operation send completion */
     UCP_WAKEUP_TAG_SEND    = UCS_BIT(2), /**< Tag send completion  */
     UCP_WAKEUP_TAG_RECV    = UCS_BIT(3), /**< Tag receive completion */
+    UCP_WAKEUP_STREAM_TX   = UCS_BIT(4), /**< Stream data was transferred */
```

Comment:
why not stream_send/stream_recv?

Reply:
I think signalling on any portion of stream data looks more preferable in stream semantic where data can be partially handled or coalesced. Also, there are at least 2 ways to handle received data. we can add STREAM_SEND & STREAM_RECV as well, but this mechanism is not implemented yet.

Reply:
i don't propose to change the meaning of the flag, just the name - to be consistent with TAG, and match the function name (stream_send_nb/stream_recv_nb)

Reply:
But the events don't mean request completion like TAG events assumes. And we won't be able to extent it on requests this way.

Reply:
so what is the exact meaning of these events?

Reply:
I would interpret them as the following from usage perspective:
TX: 1) it's time to call the progress if there are not completed requests, 2) for the future extension with tx-limit on sending - data can be send (like POLLOUT)
RX: it's time to recv the data by `ucp_recv_data_nb` or `ucp_stream_recv_nb` (posted before or in future)

Reply:
maybe call them UCP_WAKEUP_STREAM_IN / UCP_WAKEUP_STREAM_OUT as an analogy for POLLIN/POLLOUT?

Reply:
Good idea but it may be confused because both types (send/recv) are raising POLLIN events

Reply:
right..
@shamisp @bbenton what do you think